### PR TITLE
#151511475 Users should be authenticated via the user interface

### PR DESCRIPTION
**Pull request by gbengaoyetade**
#### What does this PR do?
Authenticates users from the front end

#### Description of Task to be completed?
Users, once they fill the signup or login in form correctly, they should be authenticated and redirected to the gaming screen.

#### How should this be manually tested?
Visit the app and try signing up or loging in

#### What are the relevant pivotal tracker stories?
Users should be authenticated via the user interface 

#### Screenshots (if appropriate)
slice

#### Questions:
None

### CODE REVIEW

**Review by segunolalive — COMMENTED** [review]
Good job 👍 

Ensure a user cannot return to the signin/signup page after logging in

File: public/js/controllers/index.js [hunk 1]
```diff
@@ -21,4 +21,51 @@ angular.module('mean.system')
...
+        if(response.token) {
+          window.localStorage.setItem('cfhToken', response.token);
+          $window.location.href='/#!/dashboard';
+        }
+       })
+       .error((error) => {
+        $scope.error = error.error;
+       });
+      };
+      $scope.validateInput = (userInput) => {
```
Comment (ibonly) [inline public/js/controllers/index.js]: Empty line between 2 function

File: public/js/controllers/index.js [hunk 2]
```diff
@@ -21,4 +21,51 @@ angular.module('mean.system')
...
+          $scope.error = 'password cannot be all spaces';
+          return false;
+        }
+        if (userInput.password.length < 6) {
+          $scope.error = 'password cannot be less than 6 characters';
+          return false;
+        }
+        return true;
+      }
+      $scope.signup = (userInput) => {
```
Comment (ibonly) [inline public/js/controllers/index.js]: Empty line between 2 function

**Review by ibonly — CHANGES_REQUESTED** [review]
Your PR lack test, update the test.

File: public/js/controllers/index.js (hunk 2 above)
Comment (gbengaoyetade) [reply public/js/controllers/index.js]: Done

**Comment by gbengaoyetade** [comment]
Tests written by @enodi covers this PR

**Review by ibonly — APPROVED** [review]

**Review by enodi — APPROVED** [review]

**Pull request merged**